### PR TITLE
Stop remediation when node not found

### PR DESCRIPTION
**Pull request by mshitrit**
[ECOPROJECT-1492](https://issues.redhat.com//browse/ECOPROJECT-1492)

Incorrect Status Conditions in case non existed node in Remediation CR .
When SNR is created for a non existing node Processing condition is True and Reason is RemediationStarted. 
Expected values are processing = false, succeeded = false with reason NodeNotFound.

### CODE REVIEW

**Comment by clobrano** [comment]
/lgtm

Giving a chance to get more feedbacks, feel free to unhold
/hold

File: controllers/selfnoderemediation_controller.go [hunk 1]
```diff
@@ -75,9 +75,10 @@ var (
 type processingChangeReason string
 
 const (
-	remediationStarted         processingChangeReason = "RemediationStarted"
-	remediationTerminatedByNHC processingChangeReason = "RemediationStoppedByNHC"
-	remediationFinished        processingChangeReason = "RemediationFinished"
+	remediationStarted              processingChangeReason = "RemediationStarted"
+	remediationTerminatedByNHC      processingChangeReason = "RemediationStoppedByNHC"
```
Comment (razo7) [inline controllers/selfnoderemediation_controller.go:79]: NIT: I would follow the pattern of the const name match its value by updating the const name to be `remediationStoppedByNHC `

**Comment by razo7** [comment]
NIT: Whenever we use `verifyTypeConditions` (in the UT file) the expected reason is explicitly written. Why not use the const from `selfnoderemediation_controller.go` -> Use `RemediationStarted` const instead of `"RemediationStarted"`

File: controllers/selfnoderemediation_controller.go (hunk 1 above)
Comment (slintes) [reply controllers/selfnoderemediation_controller.go:79]: and NHC is signalling a timeout actually, not stopping or terminating anything 😉

**Comment by slintes** [comment]
> NIT: Whenever we use `verifyTypeConditions` (in the UT file) the expected reason is explicitly written. Why not use the const from `selfnoderemediation_controller.go` -> Use `RemediationStarted` const instead of `"RemediationStarted"`

when you reuse the same const, you don't detect typos in const values

File: controllers/selfnoderemediation_controller.go (hunk 1 above)
Comment (razo7) [reply controllers/selfnoderemediation_controller.go:79]: Too many verbs for one action :D

**Pull request merged**